RHDEVDOCS-3849 - Added release notes for 1.5.0 (#44630)
openshift/openshift-docs: modules/gitops-release-notes-1-5-0.adoc

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc

[id="gitops-release-notes-1-5-0_{context}"]
= Release notes for {gitops-title} 1.5.0

[role="_abstract"]

> {gitops-title} 1.5.0 is now available on {product-title} 4.7, 4.8, 4.9 and 4.10.

Reviewer: @wtam2018 OCP 4.7 support is been removed right?

Reviewer 2: right

[id="new-features-1-5-0_{context}"]
== New features

The current release adds the following improvements.

* This enhancement upgrades Dex to version *2.30.3*. link:https://issues.redhat.com/browse/GITOPS-1850[GITOPS-1850]

> * This enhancement upgrades Argo CD to version *2.3.3*. link:https://issues.redhat.com/browse/GITOPS-1708[GITOPS-1708]

Reviewer 3: Can we start with this as point no.1 instead of Dex upgrade.

* With this update, the new CSV adds the annotation `olm.skipRange: '>=1.0.0 <1.5.0'` which means all the previous release versions will be skipped. The operator will upgrade to v1.5.0 directly. link:https://issues.redhat.com/browse/GITOPS-1787[GITOPS-1787]

* With this update, the RH-SSO version shipped with {gitops-title} is updated to version v7.5.1. The new RH-SSO allows you to login into their OpenShift clusters using the `kube:admin user`.  Another enhancement of the new RH-SSO is to honor the `HTTP_PROXY` environment variables. {gitops-title} propagates `HTTP_PROXY` to the RH-SSO workloads. The new RH-SSO can also configure Argo CD RBAC using OpenShift groups. link:https://issues.redhat.com/browse/GITOPS-1330[GITOPS-1330]

* With this update, a new URL field `.host` is added to the `.status` of the Argo CD. When route or ingress is enabled (priority given to route) then the route will be displayed in the new URL field. If no URL is provided from the route or ingress, the `.host` field will not be displayed.    
+

> On a non-OpenShift cluster (meaning the Route API is not available), if you choose to enable the route, a log appears saying that the routes are not available in non-OpenShift environments and to please use ingresses instead. The state of the application controller and of the URL will not be affected.

Reviewer 4: I wasn't sure if that would ever happen, I was just trying to say in general if someone tries to enable Route on a non-openshift cluster they will get the message

+

> When the route or ingress is configured, but the corresponding controller is not set up properly (i.e. is not in Ready state or does not propagate its URL), this is indicated in the Operand as well in the value of `.status.url` as `Pending` instead of the URL. Also, if `.status.url` is `Pending`, this affects the overall status for the Operand by making it `Pending` instead of `Available`. link:https://issues.redhat.com/browse/GITOPS-654[GITOPS-654]

Reviewer: It should be **status.host** instead of **.status.url**, correct me if I am wrong @reginapizza

Reviewer 4: Yes, `status.host`!

* Before this update, the TLS certificate in the `argocd-tls-certs-cm` configuration map was deleted by the {gitops-title} unless the certificate was configured in the ArgoCD CR specification `tls.initialCerts` field. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-1725[GITOPS-1725]

* Before this update, while creating a namespace with the `manage-by` label it created a lot of `RoleBinding` on the new namespace. This update fixes the issue of {gitops-title} removes the extraneous `Role` and `RoleBinding` resources that were created by the previous versions.
 link:https://issues.redhat.com/browse/GITOPS-1550[GITOPS-1550]

* Before the update,  Firefox 94 and later failed to connect to Argo CD UI with error code *SEC_ERROR_BAD_DER* since TLS certificate of the Route in pass-through mode did not not had a CA name.

> With this update, first delete the `<argocd-instance-name>-tls` secrets once the {gitops-title} is recreated. After recreating again by the {gitops-itle}, the Argo CD UI is accessible by Firefox. link:https://issues.redhat.com/browse/GITOPS-1548[GITOPS-1548]

Reviewer 3: First, delete openshift-gitops-ca, let it recreate, then delete openshift-gitops-tls, then it works. 

@DebarghoGhosh I have commented on the JIRA to correct the release notes accordingly.

[id="gitops-release-notes-1-5-0_{context}"]
= Release notes for {gitops-title} 1.5.0

[role="_abstract"]
{gitops-title} 1.5.0 is now available on {product-title} 4.8, 4.9 and 4.10.

[id="new-features-1-5-0_{context}"]
== New features

The current release adds the following improvements.

* This enhancement upgrades Argo CD to version *2.3.3*. link:https://issues.redhat.com/browse/GITOPS-1708[GITOPS-1708]

* This enhancement upgrades Application Set to version *0.4.1*. link:https://issues.redhat.com/browse/GITOPS-1850[GITOPS-1850]

* This enhancement upgrades Dex to version *2.30.3*. link:https://issues.redhat.com/browse/GITOPS-1850[GITOPS-1850]

* This enhancement upgrades Helm to version *3.8.0*. link:https://issues.redhat.com/browse/GITOPS-1709[GITOPS-1709]

* This enhancement upgrades Kustomize to version *4.4.1*. link:https://issues.redhat.com/browse/GITOPS-1710[GITOPS-1710]

* With this update, a new channel as *latest* has been added where all the latest version releases will be updated. For GitOps v1.5.0, the operator will be pushed to gitops-1.5, *latest* channel, and the existing *stable* channel. link:https://issues.redhat.com/browse/GITOPS-1791[GITOPS-1791]

> 

Reviewer 2: We should mention that we won't be pushing to the "stable" channel starting in v1.6. (We want to give users some advance notice.)

[id="fixed-issues-1-5-0_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

* Before this update, RBAC rules specific to *AppProjects* would not allow the use of commas for the role's subject field, thus preventing bindings to LDAP account. This has been fixed and complex role bindings can now be specified in *AppProject* specific RBAC rules. link:https://issues.redhat.com/browse/GITOPS-1771[GITOPS-1771]

* Before this update, when a `DeploymentConfig` is scaled to `0`, ArgoCD used to display it in *progressing* state with health status message as *"replication controller is waiting for pods to run"*. With this update, the edge case is fixed and the health check will report the correct health status of `DeploymentConfig` resource. link:https://issues.redhat.com/browse/GITOPS-1738[GITOPS-1738]

* Before this update, the TLS certificate in the `argocd-tls-certs-cm` configuration map was deleted by the {gitops-title} unless the certificate was configured in the ArgoCD CR specification `tls.initialCerts` field. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-1725[GITOPS-1725]

> * Before this update, while creating a namespace with the `manage-by` label it created a lot of `RoleBinding` on the new namespace. This update fixes the issue of {gitops-title} removes the extraneous `Role` and `RoleBinding` resources that were created by the previous versions.

Reviewer: @DebarghoGhosh Please change **manage-by** to **managed-by**

* With this update, the new CSV adds the annotation `olm.skipRange: '>=1.0.0 <1.5.0'` which means all the previous release versions will be skipped. The operator will upgrade to v1.5.0 directly. link:https://issues.redhat.com/browse/GITOPS-1787[GITOPS-1787]

* With this update, the Operator updates the RH-SSO to version v7.5.1 including the following enhancements:

** Allows you to login into the Argo CD using the OpenShift credentials including `kube:admin`.
** Supports Argo CD RBAC using OpenShift groups.
** Honors `HTTP_Proxy`, RH-SSO can be used as an SSO for Argo CD running behind a proxy. 
+
link:https://issues.redhat.com/browse/GITOPS-1330[GITOPS-1330]

* With this update, a new URL field `.host` is added to the `.status` of the Argo CD. When route or ingress is enabled (priority given to route) then the route will be displayed in the new URL field. If no URL is provided from the route or ingress, the `.host` field will not be displayed.
+

> When the route or ingress is configured, but the corresponding controller is not set up properly (i.e. is not in Ready state or does not propagate its URL), this is indicated in the Operand as well in the value of `.status.host` as `Pending` instead of the URL. This affects the overall status for the Operand by making it `Pending` instead of `Available`. link:https://issues.redhat.com/browse/GITOPS-654[GITOPS-654]

Reviewer 5: ```suggestion
When the route or ingress is configured, but the corresponding controller is not set up properly and is not in the `Ready` state or does not propagate its URL, the value of the `.status.host` field in the Operand indicates as `Pending` instead of the URL. This affects the overall status of the Operand by making it `Pending` instead of `Available`. link:https://issues.redhat.com/browse/GITOPS-654[GITOPS-654]
```

[id="fixed-issues-1-5-0_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

> * Before this update, RBAC rules specific to *AppProjects* would not allow the use of commas for the role's subject field, thus preventing bindings to LDAP account. This has been fixed and complex role bindings can now be specified in *AppProject* specific RBAC rules. link:https://issues.redhat.com/browse/GITOPS-1771[GITOPS-1771]

Reviewer 5: ```suggestion
* Before this update, RBAC rules specific to *AppProjects* would not allow the use of commas for the subject field of the role, thus preventing bindings to the LDAP account. This update fixes the issue and you can now specify complex role bindings in *AppProject* specific RBAC rules. link:https://issues.redhat.com/browse/GITOPS-1771[GITOPS-1771]
```

> * Before this update, when a `DeploymentConfig` is scaled to `0`, ArgoCD used to display it in *progressing* state with health status message as *"replication controller is waiting for pods to run"*. With this update, the edge case is fixed and the health check will report the correct health status of `DeploymentConfig` resource. link:https://issues.redhat.com/browse/GITOPS-1738[GITOPS-1738]

Reviewer 5: ```suggestion
* Before this update, when a `DeploymentConfig` resource is scaled to `0`, ArgoCD displayed it in a *progressing* state with a health status message as *"replication controller is waiting for pods to run"*. This update fixes the edge case and the health check now reports the correct health status of the `DeploymentConfig` resource. link:https://issues.redhat.com/browse/GITOPS-1738[GITOPS-1738]
```

> * Before this update, the TLS certificate in the `argocd-tls-certs-cm` configuration map was deleted by the {gitops-title} unless the certificate was configured in the ArgoCD CR specification `tls.initialCerts` field. This update fixes this issue. link:https://issues.redhat.com/browse/GITOPS-1725[GITOPS-1725]

Reviewer 5: ```suggestion
* Before this update, the TLS certificate in the `argocd-tls-certs-cm` configuration map was deleted by the {gitops-title} unless the certificate was configured in the ArgoCD CR specification `tls.initialCerts` field. This issue is fixed now. link:https://issues.redhat.com/browse/GITOPS-1725[GITOPS-1725]
```

> * Before this update, while creating a namespace with the `managed-by` label it created a lot of `RoleBinding` on the new namespace. This update fixes the issue of {gitops-title} removes the extraneous `Role` and `RoleBinding` resources that were created by the previous versions.

Reviewer 5: ```suggestion
* Before this update, while creating a namespace with the `managed-by` label it created a lot of `RoleBinding` resources on the new namespace. This update fixes the issue and now {gitops-title} removes the irrelevant `Role` and `RoleBinding` resources created by the previous versions.
```

 link:https://issues.redhat.com/browse/GITOPS-1550[GITOPS-1550]

> * Before the update,  Firefox 94 and later failed to connect to Argo CD UI with error code *SEC_ERROR_BAD_DER* since TLS certificate of the Route in pass-through mode did not not had a CA name. link:https://issues.redhat.com/browse/GITOPS-1548[GITOPS-1548]

Reviewer 5: ```suggestion
* Before this update, the TLS certificate of the route in pass-through mode did not have a CA name.  As a result, Firefox 94 and later failed to connect to Argo CD UI with error code *SEC_ERROR_BAD_DER*. 
```

+

> With this update, first delete the `<openshift-gitops-ca>` secrets, let it recreate and then delete `<openshift-gitops-tls>`. After recreating again by the {gitops-title}, the Argo CD UI is accessible by Firefox. link:https://issues.redhat.com/browse/GITOPS-1548[GITOPS-1548]

Reviewer 5: Suggestion: Move it up and align it with the previous paragraph.

Author: both are getting created by the Operator. SMEs already checked and based on their suggestion the content was changed
